### PR TITLE
Simplified for loop and also added comments in htab_hash() function in tvm_htab.c

### PR DESCRIPTION
**Pull request by dragon540**
simplified for loop in htab_hash() function also added some comments, to make it clear what it is doing. Please check this as soon as you can.

### CODE REVIEW

**Comment by dragon540** [comment]
@jakogut please let me know whether you are committing this or I whether i have done something wrong like code style or something.

File: libtvm/tvm_htab.c [hunk 1]
```diff
@@ -41,7 +41,13 @@ static inline unsigned int htab_hash(const char *k, const unsigned int size)
 {
 	unsigned int hash = 1;
 
-	char *c; for (c = (char *)k; *c; c++)
```
Comment (jakogut) [inline libtvm/tvm_htab.c:44]: Although the original code is not the clearest, I don't think this is an objective improvement.

File: libtvm/tvm_htab.c [hunk 2]
```diff
@@ -41,7 +41,13 @@ static inline unsigned int htab_hash(const char *k, const unsigned int size)
...
+	char *c = (char*)k;
+	// loop continues until ascii value at address c is not 0
```
Comment (jakogut) [inline libtvm/tvm_htab.c:45]: This is obvious by reading the code.

File: libtvm/tvm_htab.c [hunk 3]
```diff
@@ -41,7 +41,13 @@ static inline unsigned int htab_hash(const char *k, const unsigned int size)
...
+		for(; *c != 0; c++)
+
+		/* hash is incremented by (hash 'times' 2^*c)-*c on every iteration
+		 * *c is the ascii value of char stored at address c
+		 */
```
Comment (jakogut) [inline libtvm/tvm_htab.c:50]: Same here, this comment is unnecessary. The code says the same thing the comment does, with less space.

**Review by jakogut — COMMENTED** [review]
Overall, I don't think these changes are an improvement. However, a while ago, I did make the decision to switch to the Linux Kernel's style guidelines, and I'm sure there are still areas that can be improved there. You can run this code through `checkpatch.pl` for style errors, and submit patches based on that.

Thanks for your contribution! There are other suggestions for contributions in the project's TODO file.

**Comment by dragon540** [comment]
OK no problem I will see if I can contribute in some other ways.

**Pull request closed without merging**